eisen_extras/torchvision/transforms.py

```diff
@@ -0,0 +1,214 @@
+import numpy as np
+import torch
+from typing import List
+
+from PIL.Image import Image
+from torchvision import transforms as tvt
+
+
+class PILImageTorchvisionTransforms:
+    """# TODO Torchvision transforms for PIL Image input.
+    """
+    def __init__(self, fields: List[str], transform):
+        self.fields = fields
+        self.transform = transform
+
+    def __call__(self, data: dict) -> dict:
+        for field in self.fields:
+            self._check_data_type(data[field])
+            data[field] = self.transform(data[field])
+        return data
+
+    def _check_data_type(self, instance):
+        """# TODO"""
+        if not isinstance(instance, Image):
```

Comment:
I am unsure if all the expected types for torchvision transforms are PIL Image. torchvision transforms might require some other types. One thing one could do is to have the type defined during init

__init__(self):
    self.expected_type = Image

_check_data_type(self, instance):
   ...
   if not isinstance(instance, self.expected_type)
   ...

Reply:
Good point. Not sure if you saw that there is also another parent class below for Tensor input. We have two parent classes here for transforms which need PIL Image input and another class for torch.*Tensor input. Do you mean here we let the user input the data type during init of the transform or  do we use a single parent class for all the transforms? :)

Reply:
I think it's fine in the way you are doing it :) I didn't see there were two base classes

Reply:
we can also store the expected type during init but without asking the user to supply it. it would belong to the class and actually be hardcoded in it.

Reply:
Sounds good, I'll include the expected_type variable then.